packages/libs/lambda-at-edge/src/regeneration-handler.ts

```diff
@@ -0,0 +1,86 @@
+import lambdaAtEdgeCompat from "@sls-next/next-aws-cloudfront";
+import { OriginRequestDefaultHandlerManifest } from "./types";
+import { S3Client } from "@aws-sdk/client-s3";
+import { buildS3RetryStrategy } from "./s3/s3RetryStrategy";
+
+export const handler: AWSLambda.SQSHandler = async (event) => {
+  await Promise.all(
+    event.Records.map(async (record) => {
+      const bucketName = record.messageAttributes.BucketName?.stringValue;
+      const bucketRegion = record.messageAttributes.BucketRegion?.stringValue;
+      const manifestString = record.messageAttributes.Manifest?.stringValue;
+      const basePath = record.messageAttributes.BasePath?.stringValue;
+      const cloudFrontEventRequestString =
+        record.messageAttributes.CloudFrontEventRequest?.stringValue;
+      if (
+        !bucketName ||
+        !bucketRegion ||
+        !cloudFrontEventRequestString ||
+        !manifestString
+      ) {
+        throw new Error(
+          "Expected BucketName, BucketRegion, CloudFrontEventRequest & EnableHTTPCompression message attributes"
+        );
+      }
+      const cloudFrontEventRequest: AWSLambda.CloudFrontRequest = JSON.parse(
+        cloudFrontEventRequestString
+      );
+      const manifest: OriginRequestDefaultHandlerManifest = JSON.parse(
+        manifestString
+      );
+
+      const s3 = new S3Client({
```

Comment:
I know we are just on lambda-at-edge not but since this is new, if possible maybe try to separate the AWS/CloudFront specific code from the regeneration code. At least maybe just separating via different files for now, and Rollup.js can do the right bundling as needed? That way it is easier to adapt to other platforms if needed in the future (that is kind of the long term vision).

For example you can have the following:
* Core regeneration code (generic)
* Lambda wrapper (hooks up SQS event data -> core code)
* Storage client (in core code, it is abstract interface and you can write an implementation in S3 that can be used if on AWS)

Reply:
Makes a lot of sense.